cg/meta/archive/archive.py

```diff
@@ -0,0 +1,42 @@
+import logging
+from collections import namedtuple
+from typing import Dict, List
+
+
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.meta.archive.ddn_dataflow import DDNDataFlowApi
+from cg.store import Store
+
+LOG = logging.getLogger(__name__)
+
+PathAndSample = namedtuple("PathAndSample", "path sample_internal_id")
+
+
+class ArchiveAPI:
+    """Class handling the archiving of sample SPRING files."""
+
+    def __init__(
+        self, ddn_dataflow_api: DDNDataFlowApi, housekeeper_api: HousekeeperAPI, status_db: Store
+    ):
+        self.housekeeper_api: HousekeeperAPI = housekeeper_api
+        self.ddn_api: DDNDataFlowApi = ddn_dataflow_api
+        self.status_db: Store = status_db
+
+    def sort_files_on_archive_location(
+        self,
+        file_data: List[PathAndSample],
+    ) -> Dict[str, List[PathAndSample]]:
```

Comment:
I find the return type difficult to understand. Maybe we can add a data class which encapsulates it?

Reply:
It is a dictionary mapping each location to a list of files and samples?

Reply:
Yes we should start avoiding passing dicts around and have data classes for these things

Reply:
I re-wrote the function to take an archive location as an argument and fetch the files matching that. It removes the dict layer but will result in more database queries. I think it's fair trade off for readability though. Please have a look at `get_files_by_archive_location`